TalkAloud/TalkAloud/TagTableViewController.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  TagTableViewController.swift
+//  TalkAloud
+//
+//  Created by Justin Bengtson on 7/13/20.
+//  Copyright © 2020 Justin Bengtson. All rights reserved.
+//
+
+import UIKit
+
+protocol TagFilterDelegate: class {
+    func didUpdateTagToFilter(by tags: [String])
+}
+
+class TagTableViewController: UITableViewController {
+    
+    weak var delegate: TagFilterDelegate?
+    private var indexes = [Int]()
+    private var selectedTags = [String]()
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        AudioManager.sharedInstance.getAllAudioRecordingTags() ?? []
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return AudioManager.sharedInstance.getAllTagsCount()
+    }
+    
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let currentTag = AudioManager.sharedInstance.getTagForIndex(index: indexPath.row)
+        
+        let cell = tableView.dequeueReusableCell(withIdentifier: "AudioRecordingTags", for: indexPath)
+        cell.textLabel?.text = currentTag
+        return cell
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.allowsMultipleSelection = true
+        if let selectedCells = tableView.indexPathsForSelectedRows {
+            for selectedCell in selectedCells {
+                indexes.append(selectedCell.row)
+            }
+        }
+    }
+    
+    @IBAction func rightButtonAction(_ sender: Any) {
+        self.dismiss(animated: true)
```

Comment:
It's recommended to do everything before dismissing a view controller, because when you are dismissing that view controller, you are basically removing that object from memory. 
Thus you won't be 100% sure that the code that is below the dismiss is gonna work.

Reply:
Oh, super good point! Meant to do that 😅